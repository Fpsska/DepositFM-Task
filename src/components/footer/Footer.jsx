import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { setResponseInfo, setRequestInfo } from '../../app/reducers/formSlice';
import './footer.scss';

const Footer = () => {
    const { isFormPage } = useSelector(state => state.navReducer);
    const { isFormSubmited, currentResponseInfo, currentRequestInfo } = useSelector(state => state.formSlice);
    const dispatch = useDispatch();
    // 
    useEffect(() => {
        dispatch(setRequestInfo({ message: 'waiting for submit', status: 'waiting for submit' }));
        dispatch(setResponseInfo({ message: 'waiting for submit', status: 'waiting for submit' }));
    }, []);
    // 
    return (
        <footer className="footer">
            <div className="footer__wrapper">
                {isFormPage
                    ?
                    <div className="response">
                        <span className="response__text">Response</span>
                        <div className="response__information">
                            <span className="response__status">{`Status: ${isFormSubmited ? currentRequestInfo.status : currentResponseInfo.status}`}</span>
                            <span className="response__message">{`Message:  ${isFormSubmited ? currentRequestInfo.message : currentResponseInfo.message}`}</span>
                        </div>
                    </div>
                    :
                    <></>
                }
            </div>
        </footer>
    );
};

export default Footer;