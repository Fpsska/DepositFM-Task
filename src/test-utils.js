import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { BrowserRouter } from 'react-router-dom';

import navSlice from './app/slices/navSlice';
import formSlice from './app/slices/formSlice';
import paletteSlice from './app/slices/paletteSlice';

// /. imports

const render = (
    ui,
    { preloadedState, store = configureStore({ reducer: { navSlice, formSlice, paletteSlice }, preloadedState }),
        ...renderOptions } = {}
) => {

    function Wrapper({ children }) {
        return (
            <BrowserRouter>
                <Provider store={store}>{children}</Provider>
            </BrowserRouter>
        );
    }

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };