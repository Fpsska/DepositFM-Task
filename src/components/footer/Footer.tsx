import React, { useEffect } from 'react';

import { useAppDispatch, useAppSelector } from '../../app/hooks';

import { setResponseInfo } from '../../app/slices/formSlice';

import { RootState } from '../../app/store';

import './footer.scss';

// /. imports

const Footer: React.FC = () => {
    const { isFormPage } = useAppSelector((state: RootState) => state.navSlice);
    const { currentResponseInfo, isResponseIncorrect, isFormSubmited } = useAppSelector((state: RootState) => state.formSlice);
    const dispatch = useAppDispatch();
    // 
    useEffect(() => {
        dispatch(setResponseInfo(
            {
                id: Math.floor(Math.random() * 1000),
                message: 'waiting for submit',
                status: 'waiting for submit'
            }
        )
        );
    }, []);
    // 
    return (
        <footer className="footer">
            <div className="footer__wrapper">
                {isFormPage
                    ?
                    <div className="response">
                        <div className="response__wrapper">
                            <span className="response__text">Response</span>
                            <div className="response__information">

                                <div className="response__status status">
                                    <span className="status__title">Status:{' '}</span>
                                    <span className={isResponseIncorrect ? 'status__body error' : 'status__body'} data-testid="status-body">
                                        {isFormSubmited ? currentResponseInfo.status : 'waiting for submit'}
                                    </span>
                                </div>

                                <span className="response__message" data-testid="response-message">
                                    {`Message: ${isFormSubmited ? currentResponseInfo.message : 'waiting for submit'}`}
                                </span>

                            </div>
                        </div>

                    </div>
                    :
                    <></>
                }
            </div>
        </footer>
    );
};

export default Footer;