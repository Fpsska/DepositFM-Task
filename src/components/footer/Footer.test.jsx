import { render, screen } from '../../test-utils';

import Footer from './Footer';

// /. imports

describe('Footer component', () => {
    test('render component snapshot with data', () => {
        const { container } = render(<Footer />);

        expect(container).toMatchSnapshot();
    });
    test('should be have class status__body, when isResponseIncorrect is false', () => {
        render(<Footer />);

        expect(screen.getByTestId('status-body')).toHaveClass('status__body');
    });
    test('should be have `waiting for submit` text in status fileds, when isFormSubmited is false', () => {
        render(<Footer />);

        expect(screen.getByTestId('status-body')).toHaveTextContent('waiting for submit');
        expect(screen.getByTestId('response-message')).toHaveTextContent('waiting for submit');
    });
});