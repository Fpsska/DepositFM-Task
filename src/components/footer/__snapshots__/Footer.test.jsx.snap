// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component render component snapshot with data 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="footer__wrapper"
    >
      <div
        class="response"
      >
        <div
          class="response__wrapper"
        >
          <span
            class="response__text"
          >
            Response
          </span>
          <div
            class="response__information"
          >
            <div
              class="response__status status"
            >
              <span
                class="status__title"
              >
                Status:
                 
              </span>
              <span
                class="status__body"
                data-testid="status-body"
              >
                waiting for submit
              </span>
            </div>
            <span
              class="response__message"
              data-testid="response-message"
            >
              Message: waiting for submit
            </span>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
